src/main/java/com/mitchej123/hodgepodge/core/commands/DebugCommand.java

```diff
@@ -0,0 +1,76 @@
+package com.mitchej123.hodgepodge.core.commands;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+import net.minecraft.command.CommandBase;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.util.ChatComponentText;
+
+import com.mitchej123.hodgepodge.core.util.AnchorAlarm;
+
+public class DebugCommand extends CommandBase {
+    @Override
+    public String getCommandName() {
+        return "hp";
+    }
+
+    @Override
+    public String getCommandUsage(ICommandSender sender) {
+        return "Usage: hp <subcommand>. Valid subcommands are: toggle, anchor.";
+    }
+    private void printHelp(ICommandSender sender) {
+        sender.addChatMessage(new ChatComponentText("Usage: hp <toggle|anchor>"));
+        sender.addChatMessage(new ChatComponentText("\"toggle anchordebug\" - toggles RC anchor debugging"));
+        sender.addChatMessage(new ChatComponentText("\"anchor list <player>\" - list RC anchors placed by the player (empty for current player)"));
+    }
+    @Override
+    public List addTabCompletionOptions(ICommandSender sender, String[] ss) {
+        List<String> l = new ArrayList<>();
+        String test = ss.length == 0 ? "" : ss[0].trim();
+        if (ss.length == 0 || ss.length == 1 && (test.isEmpty() || Stream.of("toggle", "anchor").anyMatch(s -> s.startsWith(test)))) {
+            Stream.of("toggle", "anchor")
+                .filter(s -> test.isEmpty() || s.startsWith(test))
+                .forEach(l::add);
+        } else if (test.equals("toggle")) {
+            String test1 = ss[1].trim();
+            if (test1.isEmpty() || "anchordebug".startsWith(test1))
+                l.add("anchordebug");
+        } else if (test.equals("anchor")) {
+            String test1 = ss[1].trim();
+            if (test1.isEmpty() || "list".startsWith(test1))
```

Comment:
Missing a check for situation, where admin types `/hp anchor list ` then press tab, expecting to see a list of usernames as completion.

FYI use `getListOfStringsMatchingLastWord(as, MinecraftServer.getServer().getAllUsernames())` to quickly get a list of possible completions of username.

Reply:
ok